includes/class-wc-payments-account.php

```diff
@@ -689,8 +689,17 @@ public function get_recommended_payment_methods( string $country_code ): array {
 			return [];
 		}
 
+		// assuming that the site is accessible.
+		// if the helper is present, we can use it to check the feature flag from the settings.
+		// if the helper is not present, we'll just assume that the site is live.
+		$is_site_live = true;
+		if ( class_exists( 'Automattic\WooCommerce\Internal\ComingSoon\ComingSoonHelper' ) ) {
+			$coming_soon_helper = new \Automattic\WooCommerce\Internal\ComingSoon\ComingSoonHelper();
+			$is_site_live       = $coming_soon_helper->is_site_live();
```

Comment:
I chose `$coming_soon_helper->is_site_live()` because it accounts for whether the setting was present or not in older versions of WooCommerce.

In older versions of WooCommerce, the `ComingSoonHelper` class didn't exist. I think that we can just assume the site is "live".